extension/spark-doris-connector/src/main/scala/org/apache/doris/spark/rdd/ScalaValueReader.scala

```diff
@@ -46,6 +47,9 @@ class ScalaValueReader(partition: PartitionDefinition, settings: Settings) {
   protected val logger = Logger.getLogger(classOf[ScalaValueReader])
 
   protected val client = new BackendClient(new Routing(partition.getBeAddress), settings)
+  protected val clientLock =
+    if (deserializeArrowToRowBatchAsync) new ReentrantLock()
```

Comment:
I have a question， deserializeArrowToRowBatchAsync is init below,  whether the use is valid here

Reply:
Is valid, equals to c++:
```c++
bool deserializeArrowToRowBatchAsync() {
  return ...
}
```

Reply:
My fault, `protected var deserializeArrowToRowBatchAsync` should be `protected def deserializeArrowToRowBatchAsync`